components/sysutils/mdata-client/Makefile

```diff
@@ -0,0 +1,41 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2019 Joyent, Inc.
+#
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		mdata-client
+COMPONENT_VERSION=	20190228
+COMPONENT_SUMMARY=	Cross-platform metadata client tools for use in Joyent Triton guests (both Zones and KVM)
+COMPONENT_PROJECT_URL=	https://github.com/joyent/mdata-client
+COMPONENT_FMRI=		system/virtualization/mdata-client
+COMPONENT_CLASSIFICATION= Applications/System Utilities
+COMPONENT_SRC=		$(COMPONENT_NAME)-release-$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.gz
+COMPONENT_ARCHIVE_URL=	https://github.com/joyent/mdata-client/archive/release-$(COMPONENT_VERSION).tar.gz
+COMPONENT_ARCHIVE_HASH=	sha256:1165ddb2c81139320e184577889ce3548416aef4835e26ad39d8e0f2d9f28135
+COMPONENT_LICENSE=	MIT
+COMPONENT_LICENSE_FILE=	$(COMPONENT_NAME).license
+
+include $(WS_MAKE_RULES)/prep.mk
+include $(WS_MAKE_RULES)/justmake.mk
+include $(WS_MAKE_RULES)/ips.mk
+
+build:		$(BUILD_32)
+
+install:	$(INSTALL_32)
+
+test:		$(NO_TESTS)
+
+REQUIRED_PACKAGES += system/library
```

Comment:
Also run `gmake REQUIRED_PACKAGES`, so it's clear in the Makefile, what dependencied the build system is able to find on it's own. You'll get something like:
```Makefile
# Auto-generated dependencies
REQUIRED_PACKAGES += ...
```
Whatever is needed, but not found automatically, move above the comment.

Reply:
> Can you set/overwrite $(CC)

Done!  The software appears to build correctly, now using a fully qualified `/usr/gcc/6/bin/gcc`.

> please do "gmake sample-manifest" and add resulting file

I will do this, but it's not immediately clear to me what this file is for!  Can you point me to a document that explains it?

> run `gmake REQUIRED_PACKAGES`, so it's clear in the Makefile, what dependencied the build system is able to find on it's own

Done!  It's still just `system/library` but I now have the `# Auto-generated dependencies` banner.

Reply:
> > please do "gmake sample-manifest" and add resulting file
> 
> I will do this, but it's not immediately clear to me what this file is for! Can you point me to a document that explains it?

It's a good starting point for the actual mdata-client.p5m component file and on version update one can see via `git diff`, what new files were installed (or were removed) and decide if they should be put into the package.

E.g. see https://github.com/OpenIndiana/oi-userland/pull/4818/files. On version update three new files emerged in the sample manifest (reflects the install area) and then I decided to put them to the right component p5m file.

> Done! It's still just `system/library` but I now have the `# Auto-generated dependencies` banner.

That banner is a good sign that someone actually run the `REQUIRED_PACKAGES` target and that the list of required packages is more or less complete.